update_secrets.py

```diff
@@ -0,0 +1,135 @@
+            # we need stuff in the environment for docker
+            os.environ.update(out)
+        self.envs = os.environ
+        self.configs = yaml.load(open(envfile).read())
+        self.configs = self._update_with_envs(self.configs, self.envs)
+
+    def _update_with_envs(self, configfile, env):
+        for service in configfile:
+            for env_ in configfile[service]['environment']:
+                if not configfile[service]['environment'].get(env_):
+                    configfile[service]['environment'][env_] = env.get(env_, '')
+                else:
+                    pattern = re.compile(r'\${.*}')
+                    env_value = configfile[service]['environment'][env_]
+                    to_replace = pattern.findall(str(env_value))
+                    for rpl in to_replace:
+                        configfile[service]['environment'][env_] = env.get(rpl[2:-1], '')
+        return configfile
+
+
+    def _run_commands(self, cmd, options=''):
+        out = ''
+        cmd = cmd.split(' ') + [options]
+        cmd.remove('')
+        print(cmd)
+        try:
+            output = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as process_error:
+            all_output = filter(None, process_error.output.split('\n'))
+            for output in all_output:
+                if not 'Error:' in output:
+                    output = 'Error: ' + str(output.strip())
+        print(output)
+        return output
+
+    def update(self, service=None):
+        stage = self.envs.get('STAGE', 'testing')
+        cmd = 'kubectl create secret generic %s-%s '
+        cmd_del = 'kubectl delete secret %s-%s'
+        if service:
+            if not self.configs.get(service):
+                print('Error: Service ["%s"] Not found' % service)
+                return
+            envs = self.configs[service]['environment']
+            cmd = cmd % (service, stage)
+            options = ''
+            for env in envs:
+                 options += '--from-literal=%s="%s" ' % (env, envs.get(env, ''))
+
+            print('Deleting old secrets for %s' % service)
+            out = self._run_commands(cmd_del % (service, stage))
+            print('Creating new secrets for %s' % service)
+            out = self._run_commands(cmd, options)
+            return
+
+        for serv in self.configs:
+            cmd = cmd % (serv, stage)
+            options = ''
+            for env in self.configs[serv]['environment']:
+                 options += '--from-literal=%s="%s" ' % (env, self.configs[serv]['environment'].get(env, ''))
+
+            print('Deleting old secrets for %s' % serv)
+            self._run_commands(cmd_del % (serv, stage))
+            print('Creating new secrets for %s' % serv)
+            self._run_commands(cmd, options)
+            cmd = 'kubectl create secret generic %s-%s '
+
+
+# ==============================================
+# CLI
+
+def _object_methods(obj):
```

Comment:
why do we need this just to run one method?

Reply:
Yes for now, but usually, we add more methods to the later